### PR TITLE
[ci] Update actions

### PR DESCRIPTION
**Pull request by spacey-sooty**

### CODE REVIEW

File: .github/workflows/lint-format.yml [hunk 1]
```diff
@@ -45,7 +45,7 @@ jobs:
       - name: Generate diff
         run: git diff HEAD > wpiformat-fixes.patch
         if: ${{ failure() }}
-      - uses: actions/upload-artifact@v3
+      - uses: actions/upload-artifact@v4
```
Comment (mcm001) [inline .github/workflows/lint-format.yml:48]: This must be in sync with download-artifact versions as well, I believe.

Comment (spacey-sooty) [reply .github/workflows/lint-format.yml:48]: https://github.com/spacey-sooty/photonvision/blob/actions-versions/.github/workflows/build.yml#L235 it already appears to be on v4

Comment (mcm001) [reply .github/workflows/lint-format.yml:48]: Horrifying. How did this code ever work.

Comment (spacey-sooty) [reply .github/workflows/lint-format.yml:48]: :3

**Review by mcm001 — APPROVED** [review]
We test in prod

**Pull request merged**